### ISSUE TITLE
Integration sets config_entry directly [deprecation warning]

### ISSUE THREAD

**Issue by z-master42** [al-one/hass-xiaomi-miot#2058]
```
Logger: homeassistant.helpers.frame
Quelle: helpers/frame.py:324
Erstmals aufgetreten: 18:40:24 (1 Vorkommnisse)
Zuletzt protokolliert: 18:40:24

Detected that custom integration 'xiaomi_miot' sets option flow config_entry explicitly, which is deprecated at custom_components/xiaomi_miot/config_flow.py, line 589: self.config_entry = config_entry. This will stop working in Home Assistant 2025.12
```

**Comment by al-one** [comment]
Fixed in 1.0.4

**Comment by z-master42** [comment]
@al-one That's not true. I have tried it out. According to my reading, your code change only causes self.config_entry to be used explicitly for HA >= 2024.12.
